src/communication/xcConnection.ts

```diff
@@ -6,90 +6,107 @@ let log = require("loglevel");
 import { isDebugEnabled } from "../loggerConfiguration";
 
 export interface Connection {
-    getModel(xcApiName: string, serverUrl: string, getModelListener: (error: Error, compositionModel: CompositionModel) => void): void;
+    getCompositionModel(xcApiName: string, serverUrl: string, getModelListener: (error: Error, compositionModel: CompositionModel) => void): void;
```

Comment:
This method should return a model promise (instead of using a callback mechanism.
getCompositionModel is really an async method

Reply:
done